src/main/java/org/zone/region/flag/entity/player/move/message/greetings/GreetingsFlagType.java

```diff
@@ -45,12 +62,20 @@ public void save(@NotNull ConfigurationNode node, @Nullable GreetingsFlag save)
             node.set(null);
             return;
         }
+        node.node("GreetingsMessageDisplayType").set(save.getDisplayType().getType().getId());
+        this.saveDisplay(node, save.getDisplayType());
         Optional<Component> opMessage = save.getMessage();
         if (opMessage.isEmpty()) {
             throw new IOException("Couldn't get message");
         }
         String componentAsString = GsonComponentSerializer.gson().serialize(opMessage.get());
-        node.node("Message").set(componentAsString);
+        node.node("GreetingsMessage").set(componentAsString);
+    }
+
+    public <T extends MessageDisplay> void saveDisplay(ConfigurationNode node, T displayType)
+            throws IOException {
+        Serializable<T> type = (Serializable<T>) displayType.getType();
```

Comment:
Cast without a check of Instance of. Either your design isn't right or you forgot the instance of check

Reply:
instanceof what

Reply:
Serializable

Reply:
why?

Reply:
Im confused by your comment, why is it bad to cast it without checking the instance of it?

Reply:
where am i checking?

Reply:
Im blind. Never mind.